tests/gha-run-k8s-common.sh

```diff
@@ -69,6 +73,11 @@ function create_cluster() {
         -l eastus2 \
         -n "${rg}"
 
+    az group wait --exists \
+        -g "${rg}" \
+        --interval "${AZ_GROUP_INTERVAL}" \
+        --timeout "${AZ_GROUP_TIMEOUT}"
+
```

Comment:
This isn't the problematic RG - looking at the [failure you linked](https://github.com/kata-containers/kata-containers/actions/runs/7724334694/job/21074189970?pr=8839) in the associated issue, there are 2 errors logged but only one is fatal:

1. `ERROR: (ResourceGroupNotFound) Resource group 'kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s' could not be found.` This is logged by the cleanup step below before we actually try to create the cluster and that is fine: https://github.com/kata-containers/kata-containers/blob/7ec99ce2bbb2fc1de385d48d01b21ebd55b0c29a/tests/gha-run-k8s-common.sh#L64-L68
2. `ERROR: (ResourceGroupNotFound) Create or update resource group node-kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s failed. autorest/azure: Service returned an error. Status=404 Code="ResourceGroupNotFound" Message="Resource group 'node-kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s' could not be found."` Note the RG name is `node-...`, referring to the RG passed via `--node-resource-group`. This RG contains the actual cluster nodes, rather than the AKS instance. This is the fatal error.

I've looked internally and it seems this a known new issue and a hotfix is rolling out. I've also confirmed that the client is not supposed to create this node RG. I propose to either:

1. Do nothing and wait till the hotfix is released. I will try to find out when that is.
2. Refactor this PR to create the node RG manually ourselves regardless, but I'm not sure this is supported, and even if it is we could run into the same AKS is running into and have to setup retry logic. I prefer option 1.

Reply:
Hi @sprt !

> This isn't the problematic RG - looking at the [failure you linked](https://github.com/kata-containers/kata-containers/actions/runs/7724334694/job/21074189970?pr=8839) in the associated issue, there are 2 errors logged but only one is fatal:
> 
>     1. `ERROR: (ResourceGroupNotFound) Resource group 'kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s' could not be found.` This is logged by the cleanup step below before we actually try to create the cluster and that is fine: https://github.com/kata-containers/kata-containers/blob/7ec99ce2bbb2fc1de385d48d01b21ebd55b0c29a/tests/gha-run-k8s-common.sh#L64-L68
> 

oh, I saw the first error and concluded it was the fatal (next being like a cascaded error message). I was mislead by the `delete_cluster || true`. I will add some guards within the `delete_cluster()` to only delete resources that exist, so do not mess with the logs.

Still regarding this PR, do you think it is worthless wait for the RG be created even though it's not causing problems now? It might cause future problems... unless the `az group create` is guaranteed to return to the caller after the group is effectively created. Is it the case?
 
>     2. `ERROR: (ResourceGroupNotFound) Create or update resource group node-kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s failed. autorest/azure: Service returned an error. Status=404 Code="ResourceGroupNotFound" Message="Resource group 'node-kataCI-k8s-8839-27c908f10757-qemu-ubuntu-amd64-s' could not be found."` Note the RG name is `node-...`, referring to the RG passed via `--node-resource-group`. This RG contains the actual cluster nodes, rather than the AKS instance. This is the fatal error.
> 
> 
> I've looked internally and it seems this a known new issue and a hotfix is rolling out. I've also confirmed that the client is not supposed to create this node RG. I propose to either:
> 
>     1. Do nothing and wait till the hotfix is released. I will try to find out when that is.
> 
>     2. Refactor this PR to create the node RG manually ourselves regardless, but I'm not sure this is supported, and even if it is we could run into the same AKS is running into and have to setup retry logic.

Since that the client is not supposed to create the node RG, I suggest to not adopt 2).

Reply:
> oh, I saw the first error and concluded it was the fatal (next being like a cascaded error message). I was mislead by the `delete_cluster || true`. I will add some guards within the `delete_cluster()` to only delete resources that exist, so do not mess with the logs.

Gotcha. We could also redirect the output to /dev/null - your call.

> Still regarding this PR, do you think it is worthless wait for the RG be created even though it's not causing problems now? It might cause future problems... unless the `az group create` is guaranteed to return to the caller after the group is effectively created. Is it the case?

Personally I wouldn't add more complexity to this function for that purpose. I'm no `az` expert but it seems those commands are meant to be synchronous and `az group wait` would be for concurrent scripts - historically we haven't had issues. Maybe the folks over at [Azure/azure-cli](https://github.com/Azure/azure-cli) would be interested in adding more docs here.

Reply:
> > oh, I saw the first error and concluded it was the fatal (next being like a cascaded error message). I was mislead by the `delete_cluster || true`. I will add some guards within the `delete_cluster()` to only delete resources that exist, so do not mess with the logs.
> 
> Gotcha. We could also redirect the output to /dev/null - your call.
>

Gonna send in another PR.
 
> > Still regarding this PR, do you think it is worthless wait for the RG be created even though it's not causing problems now? It might cause future problems... unless the `az group create` is guaranteed to return to the caller after the group is effectively created. Is it the case?
> 
> Personally I wouldn't add more complexity to this function for that purpose. I'm no `az` expert but it seems those commands are meant to be synchronous and `az group wait` would be for concurrent scripts - historically we haven't had issues. Maybe the folks over at [Azure/azure-cli](https://github.com/Azure/azure-cli) would be interested in adding more docs here.

ok, I will drop this PR entirely. Much appreciated your comments here @sprt , thanks!